sway-core/src/semantic_analysis/ast_node/declaration/auto_impl.rs

```diff
@@ -520,9 +521,20 @@ where
         let mut reads = false;
         let mut writes = false;
 
+        // used to check for name collisions
+        let mut names = BTreeMap::new();
+
+        // generate code
         for r in contract_fns {
             let decl = engines.de().get(r);
 
+            // Contract methods must be unique
+            if let Some(old) = names.insert(decl.name.as_str().to_string(), decl.name.span()) {
+                return Err(handler.emit_err(
```

Comment:
It looks like this will exit if it finds one instance of duplicates. It would be a better UX if we find all of the instances of duplicates and report each of them, so that the user can see all of the errors. That is, if there are two methods named `foo` and three methods named `bar`, they should see 2 errors: one for `foo` and one for `bar`.